### ISSUE TITLE
SpyderX "Instrument Access Failed" + workaround

### ISSUE THREAD

**Issue by 9001** [eoyilmaz/displaycal-py3#143]
Hey!

First off, thanks a lot for picking this up and keeping it running :>

I briefly had access to a SpyderX, and ran into some issues with it. The instrument self-calibration would succeed, but the actual display calibration would fail right away:

```
dispcal: Error - new_disprd() failed with 'Instrument Access Failed'
```

I suspected this was related to how the driver communicates with the instrument, so I tweaked DisplayCAL to stop and show a messagebox every time it was about to run a `dispcal` or `dispread` command. It worked! Unplugged and replugged the SpyderX each time the messagebox appeared, hitting the escape-key to close it and continue...

I'll be the first to admit it's pretty terrible :grin:  but it got me through the calibration!

```patch
diff --git a/DisplayCAL/worker.py b/DisplayCAL/worker.py
index 1ed3f311..82af42f3 100644
--- a/DisplayCAL/worker.py
+++ b/DisplayCAL/worker.py
@@ -5890,2 +5890,8 @@ def exec_cmd:
     ):
+        print(f"\nCMD> {cmd} {args}\n")
+        if cmd.split('/')[-1] in ['dispcal','dispread']:
+            if sys.platform == 'linux':
+                sp.Popen(['zenity','--info',f'--text={cmd} {args}']).communicate()
+            elif sys.platform == 'win32':
+                sp.Popen(['certreq']).communicate()  # untested, but should work!
         """Execute a command.
```

So I'm guessing argyllcms forgets to issue some shutdown- or goodbye-message, leaving the SpyderX in a bad state after dispcal/dispread exits... But this repo is probably where people will bump into the issue and look for answers, so figured I'd make the issue here :>

Btw -- the same thing happens with DisplayCAL `3.9.6` on archlinux (installed with pacman) and `3.8.9.3` on windows (downloaded from displaycal.net), both using version `2.3.1` of argyllcms.

Do you think this is something we should mention to the argyllcms guys, or is there a chance it's due to how DisplayCAL uses the argyllcms tools?

**Comment by eoyilmaz** [comment]
Thanks for reporting, I'm not sure if the problem is on DisplayCAL side, what happens when you run ``dispread`` command manually? Also I previously had problems with my i1Display Pro and installing ArgyllCMS through the OSes package manager solved my problem problems, instead of using the executables directly from the ArgyllCMS web site.

**Comment by eoyilmaz** [comment]
Is there any progress on this issue, did you try reinstalling ``Argyll`` with your OSes package manager? (which should create the required ``*.rules`` files in correct places).

**Comment by 9001** [comment]
Hey :>

Unfortunately I only had access to the SpyderX for a short time, so I will probably be unable to do any more testing. Sorry!

I did try running `dispread` manually, and it failed after displaycal had been operating the SpyderX, but succeeded if the command was run after reconnecting the SpyderX. However, I regretfully did not try to run the `dispread` command several times, so it could very well have failed on the second attempt, same as with displaycal.

And I did all the archlinux tests using the distro's own argyllcms and displaycal packages, so they should have been good.
